src/encoder.rs

```diff
@@ -227,15 +227,16 @@ impl Sequence {
 
     let mut operating_point_idc: [u16; MAX_NUM_OPERATING_POINTS] =
       [0; MAX_NUM_OPERATING_POINTS];
-    let mut level: [[usize; 2]; MAX_NUM_OPERATING_POINTS] =
+    let mut level: [[u8; 2]; MAX_NUM_OPERATING_POINTS] =
       [[1, 2]; MAX_NUM_OPERATING_POINTS];
     let mut tier: [usize; MAX_NUM_OPERATING_POINTS] =
       [0; MAX_NUM_OPERATING_POINTS];
 
     for i in 0..MAX_NUM_OPERATING_POINTS {
       operating_point_idc[i] = 0;
-      level[i][0] = 1; // minor
-      level[i][1] = 2; // major
+      // default to unconstrained
+      level[i][0] = 3; // minor
+      level[i][1] = 9; // major
```

Comment:
I understand that this isn't new in your PR, but since you are touching this, I find the array syntax a bit confusing. First because it's not clear without reading the code that the minor version comes before the major one. I'd expect the opposite. Second because special-purpose levels, like reserved values and unconstrained parameters, are not numbered in the spec, so having to "translate" unconstrained parameters to "9.3" is unintuitive. And indeed it has already caused an error in the assertion below.

Reply:
I would actually prefer storing it as an index. I'll change it.

Reply:
changed